refactor(ui): replace `MatrixInput` with `Matrix` Glide Data Grid integration

hdinia opened the pull request:
ANT-1557

hdinia committed:
feat(ui): add `customColumns` prop to `Matrix`

hdinia committed:
feat(ui): add `colWidth` prop to `Matrix` for custom columns width

hdinia committed:
refactor(ui): update matrix views replacing `MatrixInput` with `Matrix`

refactor(ui): update `MiscGen` matrix
refactor(ui): update `Reserves` matrix
refactor(ui): update `Solar` matrix
refactor(ui): update `Wind` matrix
refactor(ui): update `Thermal` matrices
refactor(ui): update `Renewables` matrix
refactor(ui): update `Storages` matrices

hdinia committed:
feat(ui): add custom `rowHeaders` prop on `Matrix`

hdinia committed:
feat(ui): add `enablePercentDisplay` prop on `Matrix`

hdinia committed:
feat(ui): add `enableReadOnly` prop on `Matrix`

hdinia committed:
refactor(ui-hydro): update `HydroMatrix` to use glide data grid

hdinia committed:
refactor(ui-links): update `Links` matrices

hdinia committed:
fix(ui): prevent 0 to be processed as falsy value

hdinia committed:
feat(ui): add `calculateMatrixAggregates` and update Hydro matrices

hdinia committed:
refactor(ui-results): update `Results` view matrices

hdinia committed:
refactor(ui-results): refresh and improve `ResultsDetails`

hdinia committed:
refactor(ui-bc): update `BindingConstraints` matrices

hdinia committed:
refactor(ui): use the number of updates as the indicator for the save button instead of the number of cells edited

hdinia committed:
feat(ui): add dynamic aggregates changes at each update

hdinia committed:
refactor(ui): change position of aggregates cols and background colors

hdinia committed:
feat(ui): make the `dateTime` column sticky

hdinia committed:
feat(ui): add columns resize to `MatrixGrid`

hdinia committed:
feat(ui): change the decimal and thousand separator for number cells

hdinia committed:
refactor(ui): improve aggregates columns usage

hdinia committed:
feat(ui): update date generation using `date-fns` instead of `moment`

hdinia committed:
feat(ui): add prompt for unsaved changes when switching views

hdinia committed:
feat(ui): allow fill handle on any direction

hdinia committed:
feat(ui): add month display to daily and hourly frequencies

hdinia committed:
feat(ui): reduce cells height and add smooth scroll

hdinia committed:
feat(ui): disable copy/paste keybindings

hdinia committed:
fix(ui): prevent regex DoS vulnerability in `formatNumber` function

hdinia committed:
test(ui): enhance `utils` tests and add `MatrixActions` component tests

skamril commented on webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Storages/Matrix.tsx:
```diff
@@ -12,138 +12,147 @@
-        <Tab label={t("study.modelization.storages.ruleCurves")} />
-        <Tab label={t("study.modelization.storages.inflows")} />
+    <Box sx={{ display: "flex", flexDirection: "column", width: 1, height: 1 }}>
+      <Tabs value={activeTab} onChange={handleTabChange} sx={{ width: 1 }}>
```
Better to use `TabsView`

hdinia replied:
I tried, it breaks the styles for some reason, see:
![image](https://github.com/user-attachments/assets/b05f0b4c-c3ed-426d-93ba-833d56f2bd3a)

skamril replied:
Alright, I will check it

skamril commented on webapp/src/components/App/Singlestudy/explore/Modelization/Links/LinkView/LinkMatrixView.tsx:
```diff
@@ -12,99 +12,139 @@
-    t("study.modelization.links.matrix.columns.pShiftMin"),
-    t("study.modelization.links.matrix.columns.pShiftMax"),
-  ];
+interface SingleMatrixContent {
```
Duplicate types

skamril commented on webapp/src/components/App/Singlestudy/explore/Modelization/Links/LinkView/LinkMatrixView.tsx:
```diff
@@ -12,99 +12,139 @@
-            columnsNames={columnsNames}
-            computStats={MatrixStats.NOCOL}
+    <Box sx={{ display: "flex", flexDirection: "column", width: 1, height: 1 }}>
+      <Tabs value={activeTab} onChange={handleTabChange} sx={{ width: 1 }}>
```
Better to use `TabsView`

skamril commented on webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/ResultFilters.tsx:
```diff
@@ -0,0 +1,157 @@
+}: Props) {
+  const { t } = useTranslation();
+
+  const controls = [
```
`controls`-> `filters`

skamril commented on webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/ResultFilters.tsx:
```diff
@@ -0,0 +1,157 @@
+  const controls = [
+    {
+      label: `${t("study.results.mc")}:`,
+      control: (
```
`control` -> `field`

skamril commented on webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx:
```diff
@@ -267,188 +234,82 @@ function ResultDetails() {
+            response={synthesisRes}
+            ifPending={() => <Skeleton sx={{ height: 1, transform: "none" }} />}
+            ifResolved={(matrix) =>
+              matrix && (
```
`matrix` can be empty?

hdinia replied:
yes its necessary

skamril commented on webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx:
```diff
@@ -267,188 +234,82 @@ function ResultDetails() {
+            response={mergeResponses(outputRes, matrixRes)}
+            ifPending={() => <Skeleton sx={{ height: 1, transform: "none" }} />}
+            ifResolved={([, matrix]) =>
+              matrix && (
```
Same

skamril commented on webapp/src/components/common/MatrixGrid/Matrix.tsx:
```diff
@@ -23,26 +23,45 @@ import { StudyMetadata } from "../../../common/types";
   enableTimeSeriesColumns?: boolean;
-  enableAggregateColumns?: boolean;
+  aggregateColumns?: AggregateConfig;
+  enableRowHeaders?: boolean;
```
`enable` is not necessary:

`enableRowHeaders` -> `rowHeaders`
`enableReadOnly` -> `readOnly`
`enablePercentDisplay` -> `showPercent`

`<Matrix enableRowHeaders enableReadOnly enablePercentDisplay />`

vs 

`<Matrix rowHeaders readOnly showPercent />`

skamril commented on webapp/src/components/common/MatrixGrid/index.tsx:
```diff
@@ -32,28 +33,31 @@ export interface MatrixGridProps {
   onCellEdit?: (update: GridUpdate) => void;
   onMultipleCellsEdit?: (updates: GridUpdate[]) => void;
-  readOnly?: boolean;
+  isReadOnlyEnabled?: boolean;
```
`Enabled` not necessary:

`isReadOnlyEnabled` -> `isReadOnly`
...

skamril commented on webapp/src/components/common/MatrixGrid/types.ts:
```diff
@@ -22,36 +22,90 @@ import {
 ////////////////////////////////////////////////////////////////
 
-export const ColumnTypes = {
+export const Column = {
```
These are not types, you can move it in a constants.ts file instead

skamril commented on webapp/src/components/common/MatrixGrid/useMatrix.ts:
```diff
@@ -53,27 +72,72 @@ export function useMatrix(
+  );
 
   useEffect(() => {
     fetchMatrix();
```
Better to use deps here instead of creating a `useCallback` for `fetchMatrix`

hdinia replied:
yes but its used elsewhere in the hook that's why

skamril replied:
Yes but it just call, `useCallback` still not necessary

hdinia replied:
Looking at it closer `useCallback` is indeed not necessary. A brief explanation like "not needed since function is only invoked internally, not used as prop/dependency" would make the review more valuable.

hdinia committed:
refactor(ui): improves `Matrix` types and variables naming

refactor(ui): rename variables and add minor improvements

hdinia committed:
feat(ui-debug): replace `MatrixInput` and add disable import prop

hdinia committed:
fix(ui): resolve imports errors due to merge conflicts


fix: correct lint and types issues 


fix(ui): resolve merge conflict

hdinia committed:
fix(ui): correct display condition in `Matrix` to handle empty subarray presence correctly

hdinia committed:
feat(ui): enable aggregates on thermal and renewables clusters

hdinia committed:
feat(ui): add `useMatrixPortal`to manage matrices on split views


feat(ui): update `useMatrixPortal` and improve test coverage

hdinia committed:
fix(ui): disable percent display on allocation and correlation matrices

hdinia committed:
test(ui): improve `Matrix` test suite organization and maintainability


test(ui): improve MatrixActions test suite organization

test(ui): improve and fix type issues in useColumnMapping tests

test(ui): improve useGridCellContent test organization

test(ui): fix async handling in useMatrix tests

test(ui): improve utils test suite organisation and typing

hdinia committed:
refactor(ui): remove `MatrixInput`

hdinia committed:
refactor(ui): improves `Matrix` folder and files overall organization

hdinia committed:
build(vite): exclude test files from production builds

sylvlecl reviewed (changes requested):
Change request just to make sure we don't forget to revert the installer submodule update

hdinia committed:
refactor(ui): rename core folder

hdinia committed:
refactor(ui): disable percents on daily power and reservoir levels

hdinia committed:
fix(ui): prevent undefined error in number formatting

hdinia committed:
feat(ui-hydro): add data adapter for `Allocation` and `Correlation`

Temporary solution to convert API numeric arrays to string arrays for Matrix 
component headers. Will be removed when API model is updated to return proper 
string labels.
